src/utils/createComponentWithOrderedProps.ts

```diff
@@ -14,7 +14,9 @@ export default function createComponentWithOrderedProps<P extends {}, E extends
     }
 
     for (const name of names) {
-      orderedProps[name] = props[name];
+      if (props.hasOwnProperty(name)) {
```

Comment:
```suggestion
      if (Object.hasOwn(props, name)) {
```
There are some [caveats](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#description) with `hasOwnProperty`, so recently the static `hasOwn` method was introduced. It is supported by all evergreen browsers, and is recommended by the W3C commetee.

Reply:
Didn't notice it's a cherry-pick. Let's have another PR for this fix.